### PR TITLE
simplify caller interface

### PR DESCRIPTION
**Pull request by dustinxie**
1. removed unnecessary interfaces
2. merged common methods to `SendActionCaller` interface

### CODE REVIEW

File: iotex/caller_claimreward.go [hunk 1]
```diff
@@ -11,56 +11,29 @@ import (
...
-	return c
-}
-
-func (c *claimRewardCaller) SetNonce(n uint64) ClaimRewardCaller {
-	c.nonce = n
-	return c
-}
-
-func (c *claimRewardCaller) API() iotexapi.APIServiceClient { return c.api }
-
```
Comment (dustinxie) [inline iotex/caller_claimreward.go:51]: common codes, move to `sendActionCaller`

File: iotex/interfaces.go [hunk 2]
```diff
@@ -77,28 +66,19 @@ type ReadContractCaller interface {
...
 }
 
 // DeployContractCaller is used to perform a deploy contract call.
 type DeployContractCaller interface {
-	SendActionCaller
-
+	Caller
 	SetArgs(abi abi.ABI, args ...interface{}) DeployContractCaller
-	SetGasPrice(*big.Int) DeployContractCaller
-	SetGasLimit(uint64) DeployContractCaller
```
Comment (dustinxie) [inline iotex/interfaces.go:94]: duplicate `SetXXX` methods are merged to `SendActionCaller`

File: iotex/interfaces.go [hunk 3]
```diff
@@ -12,30 +12,26 @@ import (
...
-}
-
-// ClaimRewardCaller is used to perform a claim reward call.
-type ClaimRewardCaller interface {
-	SendActionCaller
-
-	SetGasPrice(*big.Int) ClaimRewardCaller
-	SetGasLimit(uint64) ClaimRewardCaller
-	SetData([]byte) ClaimRewardCaller
-	SetNonce(uint64) ClaimRewardCaller
```
Comment (dustinxie) [inline iotex/interfaces.go:38]: after merging duplicated methods, these 2 are no longer needed

File: iotex/interfaces.go [hunk 4]
```diff
@@ -12,30 +12,26 @@ import (
...
+// SendActionCaller is used to set nonce/gas and derive other callers.
+type SendActionCaller interface {
```
Comment (Liuhaai) [inline iotex/interfaces.go:22]: this doesn't look like a base class. Unshared methods should be removed

File: iotex/interfaces.go [hunk 5]
```diff
@@ -109,26 +108,17 @@ type ReadOnlyContract interface {
...
-	Update(name string, operatorAddr, rewardAddr address.Address) StakingAPICaller
-}
-
-// StakingAPICaller is used to perform extra info call.
-type StakingAPICaller interface {
-	SendActionCaller
-	SetGasPrice(*big.Int) StakingAPICaller
-	SetGasLimit(uint64) StakingAPICaller
-	SetNonce(uint64) StakingAPICaller
-	SetPayload([]byte) StakingAPICaller
```
Comment (dustinxie) [inline iotex/interfaces.go:133]: this no longer needed

**Review by Liuhaai — APPROVED** [review]

**Pull request merged**